### DISCUSSION TITLE
How to fill the privateKey attribute?

### DISCUSSION THREAD

**Discussion by nicolascbarbosa** [renovatebot/renovate#8432] [Request Help] [answered]
## Which Renovate are you using?
Renovate Open Source CLI

## Which platform are you using?
Bitbucket Cloud

## Have you checked the logs? Don't forget to include them if relevant
This is the command I run

```
// In my package.json
"renovate":  "renovate  --password=$PASSWORD --username=$USERNAME --platform=bitbucket some-org/some-repo"

yarn renovate
```

And this is the error I get
```
 INFO: Repository started (repository=some-org/some-repo)
       "renovateVersion": "24.28.1"
ERROR: Found encrypted data but no privateKey (repository=some-org/some-repo)
 INFO: Dependency extraction complete (repository=some-org/some-repo)
       "baseBranch": "master",
       "stats": {
         "managers": {"npm": {"fileCount": 1, "depCount": 27}},
         "total": {"fileCount": 1, "depCount": 27}
       }
 WARN: No github.com token has been configured. Skipping release notes retrieval (repository=some-org/some-repo)
       "manager": "npm",
       "depName": "web-vitals",
       "sourceUrl": "https://github.com/GoogleChrome/web-vitals"
 WARN: No github.com token has been configured. Skipping release notes retrieval (repository=some-org/some-repo)
       "manager": "npm",
       "depName": "renovate",
       "sourceUrl": "https://github.com/renovatebot/renovate"
 WARN: No github.com token has been configured. Skipping release notes retrieval (repository=some-org/some-repo)
       "manager": "npm",
       "depName": "eslint-config-prettier",
       "sourceUrl": "https://github.com/prettier/eslint-config-prettier"
 WARN: No github.com token has been configured. Skipping release notes retrieval (repository=some-org/some-repo)
       "manager": "npm",
       "depName": "eslint-import-resolver-lerna",
       "sourceUrl": "https://github.com/Dreamscapes/eslint-import-resolver-lerna"
 WARN: artifactErrors (repository=some-org/some-repo, branch=renovate/pin-dependencies)
       "artifactErrors": [
         {
           "lockFile": "yarn.lock",
           "stderr": "error An unexpected error occurred: \"Failed to replace env in config: ${NPM_TOKEN}\".\n"
         }
       ]
 INFO: Repository finished (repository=some-org/some-repo)
       "durationMs": 10241
 INFO: Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 50,
           "logContext": "Ou_RMsSH0",
           "repository": "some-org/some-repo",
           "msg": "Found encrypted data but no privateKey"
         }
       ]
error Command failed with exit code 1.
```

## What would you like to do?
We are using Nexus Repository as our private package repository.

We already have Renovate running on some repositories and it works well.

For security reasons, we generated the `.npmrc` file in the pipeline (ci / cd) in order not to git version the npm token.
In some pull requests opened by Renovate, we started to encounter the following error, e.g.

```
error An unexpected error occurred: "http: //nexus-org-url/repository/npm/eslint/-/eslint-7.18.0.tgz: Request failed \" 401 Unauthorized \ "".
```

So we conclude that Renovate is unable to read the `.npmrc` generated during the pipeline.
We would then like to use the `npmrc` attribute in the configuration file (renovate.json).

We did a test with the token in plain text and it works, when I try to use the `npmToken` encryption, it doesn't work, I have doubts on how to use the `privateKey` attribute, I'm following the guide available at https://docs.renovatebot.com/self-hosted-configuration /#privatekey

I am generating the keys, encrypting with the public key and then placing it to decrypt with the private key as a string within the `privateKey` attribute.

The closest I got to making it work was this link
https://github.com/renovatebot/config-help/issues/598

**Comment by viceice** [comment, marked as answer]
You should configure the crednetials as hostrule instead of npmrc.

**Reply by nicolascbarbosa** [reply]
Thank you very much, it was very quick :smile: 
I'll try and tell you
